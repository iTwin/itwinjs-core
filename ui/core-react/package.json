{
  "name": "@itwin/core-react",
  "version": "3.0.0-dev.151",
  "description": "iTwin.js UI core components",
  "main": "lib/cjs/core-react.js",
  "module": "lib/esm/core-react.js",
  "typings": "lib/cjs/core-react",
  "imodeljsSharedLibrary": true,
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/iTwin/itwinjs-core/tree/master/ui/core-react"
  },
  "scripts": {
    "build": "npm run -s copy:locale && npm run -s build:cjs",
    "build:ci": "npm run -s build && npm run -s build:esm",
    "build:cjs": "tsc 1>&2 --outDir lib/cjs && npm run -s copy:cjs",
    "build:esm": "tsc 1>&2 --module ES2020 --outDir lib/esm && npm run -s copy:esm",
    "copy:cjs": "cpx \"./src/**/*.{*css,svg}\" ./lib/cjs",
    "copy:esm": "cpx \"./src/**/*.{*css,svg}\" ./lib/esm",
    "copy:locale": "cpx \"./public/**/*\" ./lib/public",
    "pseudolocalize": "betools pseudolocalize --englishDir ./public/locales/en --out ./public/locales/en-PSEUDO",
    "clean": "rimraf lib .rush/temp/package-deps*.json",
    "cover": "nyc npm -s test",
    "docs": "betools docs --includes=../../generated-docs/extract --json=../../generated-docs/ui/core-react/file.json --tsIndexFile=./core-react.ts --onlyJson --excludeGlob=**/declarations.d.ts",
    "lint": "eslint -f visualstudio \"./src/**/*.{ts,tsx}\" 1>&2",
    "extract-api": "betools extract-api --entry=core-react",
    "test": "mocha --config ../.mocharc.json \"./lib/cjs/test/**/*.test.js\"",
    "test:watch": "npm -s test -- --reporter min --watch-extensions ts,tsx --watch"
  },
  "keywords": [
    "Bentley",
    "BIM",
    "iModel",
    "digital-twin",
    "iTwin"
  ],
  "author": {
    "name": "Bentley Systems, Inc.",
    "url": "http://www.bentley.com"
  },
  "peerDependencies": {
    "@itwin/appui-abstract": "workspace:^3.0.0-dev.151",
    "@itwin/core-bentley": "workspace:^3.0.0-dev.151",
    "react": "^17.0.0",
    "react-dom": "^17.0.0"
  },
  "//devDependencies": [
    "NOTE: All peerDependencies should also be listed as devDependencies since peerDependencies are not considered by npm install",
    "NOTE: All tools used by scripts in this package must be listed as devDependencies"
  ],
  "devDependencies": {
    "@itwin/appui-abstract": "workspace:*",
    "@itwin/build-tools": "workspace:*",
    "@itwin/core-bentley": "workspace:*",
    "@itwin/core-common": "workspace:*",
    "@itwin/core-i18n": "workspace:*",
    "@itwin/eslint-plugin": "workspace:*",
    "@testing-library/react": "^12.0.0",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^13.2.1",
    "@types/chai": "^4.1.4",
    "@types/chai-as-promised": "^7",
    "@types/chai-jest-snapshot": "^1.3.0",
    "@types/chai-spies": "^1.0.0",
    "@types/dompurify": "^2.0.3",
    "@types/enzyme": "3.9.3",
    "@types/lodash": "^4.14.0",
    "@types/mocha": "^8.2.2",
    "@types/node": "14.14.31",
    "@types/react": "17.0.31",
    "@types/react-autosuggest": "10.1.2",
    "@types/react-dom": "^17.0.0",
    "@types/react-select": "3.0.26",
    "@types/sinon": "^9.0.0",
    "@types/sinon-chai": "^3.2.0",
    "@wojtekmaj/enzyme-adapter-react-17": "^0.6.3",
    "chai": "^4.1.2",
    "chai-as-promised": "^7",
    "chai-jest-snapshot": "^2.0.0",
    "chai-spies": "1.0.0",
    "cpx2": "^3.0.0",
    "enzyme": "^3.4.0",
    "enzyme-to-json": "^3.3.4",
    "eslint": "^7.11.0",
    "ignore-styles": "^5.0.1",
    "jsdom": "^17.0.0",
    "jsdom-global": "3.0.2",
    "mocha": "^8.3.2",
    "nyc": "^15.1.0",
    "raf": "^3.4.0",
    "react": "^17.0.0",
    "react-dom": "^17.0.0",
    "react-dom-instance": "2.0.0",
    "react-test-renderer": "^17.0.0",
    "rimraf": "^3.0.2",
    "sinon": "^9.0.2",
    "sinon-chai": "^3.2.0",
    "ts-node": "^7.0.1",
    "typemoq": "^2.1.0",
    "typescript": "~4.4.0",
    "xmlhttprequest": "^1.8.0"
  },
  "//dependencies": [
    "NOTE: these dependencies should be only for things that DO NOT APPEAR IN THE API",
    "NOTE: core-frontend should remain UI technology agnostic, so no react/angular dependencies are allowed"
  ],
  "dependencies": {
    "@bentley/icons-generic-webfont": "^1.0.15",
    "@itwin/itwinui-css": "^0.37.0",
    "@itwin/itwinui-react": "^1.23.0",
    "classnames": "^2.3.1",
    "dompurify": "^2.0.12",
    "lodash": "^4.17.10",
    "react-autosuggest": "^10.1.0",
    "react-select": "3.2.0",
    "resize-observer-polyfill": "1.5.1"
  },
  "nyc": {
    "extends": "./node_modules/@itwin/build-tools/.nycrc",
    "require": [
      "ignore-styles",
      "jsdom-global/register",
      "source-map-support/register",
      "ts-node/register"
    ],
    "check-coverage": true,
    "statements": 100,
    "branches": 100,
    "functions": 100,
    "lines": 100
  },
  "eslintConfig": {
    "plugins": [
      "@itwin"
    ],
    "extends": [
      "plugin:@itwin/ui",
      "plugin:@itwin/jsdoc"
    ]
  }
}
