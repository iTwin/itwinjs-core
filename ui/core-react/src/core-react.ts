/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/

// cSpell:ignore focustrap imagecheckbox iconinput hocs numberinput numericinput

export * from "./core-react/UiCore";

export * from "./core-react/badge/Badge";
export * from "./core-react/badge/BadgeUtilities";
export * from "./core-react/badge/BetaBadge";
export * from "./core-react/badge/NewBadge";

export * from "./core-react/base/Div";
export * from "./core-react/base/DivWithOutsideClick";
export * from "./core-react/base/Centered";
export * from "./core-react/base/FillCentered";
export * from "./core-react/base/FlexWrapContainer";
export * from "./core-react/base/Gap";
export * from "./core-react/base/PointerEvents";
export * from "./core-react/base/ScrollView";

export * from "./core-react/button/Button";
export * from "./core-react/button/UnderlinedButton";

export * from "./core-react/checklistbox/CheckListBox";

export * from "./core-react/contextmenu/ContextMenu";
export * from "./core-react/contextmenu/ContextMenuDirection";
export * from "./core-react/contextmenu/ContextMenuDivider";
export * from "./core-react/contextmenu/ContextMenuItem";
export * from "./core-react/contextmenu/ContextSubMenu";
export * from "./core-react/contextmenu/GlobalContextMenu";
export * from "./core-react/contextmenu/PopupContextMenu";

export * from "./core-react/dialog/Dialog";
export * from "./core-react/dialog/DialogButtonDef";
export * from "./core-react/dialog/GlobalDialog";

export * from "./core-react/elementseparator/ElementSeparator";

export * from "./core-react/enums/Alignment";
export * from "./core-react/enums/CheckBoxState";
export * from "./core-react/enums/Orientation";
export * from "./core-react/enums/SortDirection";
export * from "./core-react/enums/TimeFormat";

export * from "./core-react/expandable/ExpandableList";
export * from "./core-react/expandable/ExpandableBlock";

export * from "./core-react/focus/ItemKeyboardNavigator";
export * from "./core-react/focustrap/FocusTrap";

export * from "./core-react/form/Field";
export * from "./core-react/form/Form";

export * from "./core-react/hocs/withIsPressed";
export * from "./core-react/hocs/withOnOutsideClick";
export * from "./core-react/hocs/withTimeout";

export * from "./core-react/icons/IconComponent";
export * from "./core-react/icons/SvgPath";
export * from "./core-react/icons/SvgSprite";
export * from "./core-react/icons/WebFontIcon";

export * from "./core-react/autosuggest/AutoSuggest";

export * from "./core-react/checkbox/Checkbox";

export * from "./core-react/imagecheckbox/ImageCheckBox";

export * from "./core-react/inputs/Input";
export * from "./core-react/inputs/InputLabel";
export * from "./core-react/inputs/InputStatus";
export * from "./core-react/inputs/iconinput/IconInput";
export * from "./core-react/inputs/LabeledComponentProps";
export * from "./core-react/inputs/LabeledInput";
export * from "./core-react/inputs/LabeledTextarea";
export * from "./core-react/inputs/numberinput/NumberInput";
export * from "./core-react/inputs/Textarea";

export * from "./core-react/listbox/Listbox";

export * from "./core-react/loading/LoadingBar";
export * from "./core-react/loading/LoadingPrompt";
export * from "./core-react/loading/LoadingSpinner";
export * from "./core-react/loading/LoadingStatus";
export * from "./core-react/loading/Spinner";

export * from "./core-react/messagebox/MessageBox";
export * from "./core-react/messagebox/MessageSeverity";

export * from "./core-react/notification/MessageRenderer";
export * from "./core-react/notification/MessageType";

export * from "./core-react/popup/Popup";

export * from "./core-react/progress-indicators/ProgressBar";
export * from "./core-react/progress-indicators/ProgressSpinner";

export * from "./core-react/radialmenu/RadialMenu";
export * from "./core-react/radialmenu/Annulus";
export * from "./core-react/radio/Radio";

export * from "./core-react/select/LabeledSelect";
export * from "./core-react/select/Select";
export * from "./core-react/select/ThemedSelect";
export * from "./core-react/select/LabeledThemedSelect";
export * from "./core-react/searchbox/SearchBox";

export * from "./core-react/settings/SettingsManager";
export * from "./core-react/settings/SettingsContainer";

export * from "./core-react/slider/Slider";

export * from "./core-react/splitbutton/SplitButton";

export * from "./core-react/tabs/HorizontalTabs";
export * from "./core-react/tabs/VerticalTabs";
export * from "./core-react/tabs/Tabs";

export * from "./core-react/text/BodyText";
export * from "./core-react/text/BlockText";
export * from "./core-react/text/DisabledText";
export * from "./core-react/text/FilteredText";
export * from "./core-react/text/Headline";
export * from "./core-react/text/LeadingText";
export * from "./core-react/text/LeadingText2";
export * from "./core-react/text/MutedText";
export * from "./core-react/text/SmallText";
export * from "./core-react/text/Subheading";
export * from "./core-react/text/Subheading2";
export * from "./core-react/text/StyledText";
export * from "./core-react/text/TextProps";
export * from "./core-react/text/Title";
export * from "./core-react/text/Title2";

export * from "./core-react/tiles/FeaturedTile";
export * from "./core-react/tiles/MinimalFeaturedTile";
export * from "./core-react/tiles/MinimalTile";
export * from "./core-react/tiles/Tile";

export * from "./core-react/toggle/Toggle";
export * from "./core-react/toggle/LabeledToggle";

export * from "./core-react/tooltip/Tooltip";

export { ExpansionToggle, ExpansionToggleProps } from "./core-react/tree/ExpansionToggle";
export { TreeBranch, TreeBranchProps } from "./core-react/tree/Branch";
export { TreeNode, TreeNodeProps, NodeCheckboxProps, NodeCheckboxRenderer, NodeCheckboxRenderProps } from "./core-react/tree/Node";
export { Tree, TreeProps } from "./core-react/tree/Tree";
export { TreeNodePlaceholder, TreeNodePlaceholderProps } from "./core-react/tree/Placeholder";

export * from "./core-react/uistate/UiStateEntry";
export * from "./core-react/uistate/UiStateStorage";
export * from "./core-react/uistate/LocalStateStorage";

export * from "./core-react/utils/IconHelper";
export * from "./core-react/utils/Point";
export * from "./core-react/utils/PointProps";
export * from "./core-react/utils/Props";
export * from "./core-react/utils/Rectangle";
export * from "./core-react/utils/Size";
export * from "./core-react/utils/Timer";
export * from "./core-react/utils/UiEvent";
export * from "./core-react/utils/UiGeometry";
export * from "./core-react/utils/flattenChildren";
export * from "./core-react/utils/getBestBWContrastColor";
export * from "./core-react/utils/getCssVariable";
export * from "./core-react/utils/getDisplayName";
export * from "./core-react/utils/getUserColor";
export * from "./core-react/utils/shallowDiffers";
export * from "./core-react/utils/typeUtils";
export * from "./core-react/utils/isPromiseLike";
export * from "./core-react/utils/ScrollPositionMaintainer";

export * from "./core-react/utils/hooks/useDisposable";
export * from "./core-react/utils/hooks/useEffectSkipFirst";
export * from "./core-react/utils/hooks/useEventListener";
export * from "./core-react/utils/hooks/ResizeObserverPolyfill";
export * from "./core-react/utils/hooks/useOnOutsideClick";
export * from "./core-react/utils/hooks/useProximityToMouse";
export * from "./core-react/utils/hooks/useRefEffect";
export * from "./core-react/utils/hooks/useRefs";
export * from "./core-react/utils/hooks/useRefState";
export * from "./core-react/utils/hooks/useResizeObserver";
export * from "./core-react/utils/hooks/useTargeted";
export * from "./core-react/utils/hooks/useWidgetOpacityContext";

/** @docs-package-description
 * The core-react package contains general purpose React components, such as Dialog, MessageBox, SearchBox, RadialMenu and SplitButton.
 * For more information, see [learning about core-react]($docs/learning/ui/core/index.md).
 */
/**
 * @docs-group-description AutoSuggest
 * Component for input with an auto-suggestion dropdown.
 */
/**
 * @docs-group-description Base
 * Low-level classes and components for building application UI.
 */
/**
 * @docs-group-description Button
 * Components for working with various Buttons.
 */
/**
 * @docs-group-description Checkbox
 * Component is a wrapper for the `<input type="checkbox">` HTML element.
 */
/**
 * @docs-group-description CheckListBox
 * Components for working with a Check listbox.
 */
/**
 * @docs-group-description Common
 * Common classes and enums used across various UI components.
 */
/**
 * @docs-group-description ContextMenu
 * Components for working with a Context Menu.
 */
/**
 * @docs-group-description Dialog
 * Components for working with a Dialog or MessageBox.
 */
/**
 * @docs-group-description ElementSeparator
 * Components for working with a ElementSeparator.
 */
/**
 * @docs-group-description Expandable
 * Components for working with a ExpandableBlock or ExpandableList.
 */
/**
 * @docs-group-description Form
 * Components used to create a Form using supplied properties to specify fields.
 */
/**
 * @docs-group-description Icon
 * Component that renders core-react icon when given an icon name or SVG source.
 */
/**
 * @docs-group-description Inputs
 * Components for working with input controls, such as Input, IconInput, NumberInput and Textarea.
 */
/**
 * @docs-group-description Loading
 * Components for working with Loading spinners and progress indicators and bars.
 */
/**
 * @docs-group-description Notification
 * Components for working with messages and tooltips.
 */
/**
 * @docs-group-description Popup
 * Components for working with a Popup.
 */
/**
 * @docs-group-description RadialMenu
 * Components for working with a RadialMenu.
 */
/**
 * @docs-group-description Radio
 * Component is a wrapper for the `<input type="radio">` HTML element.
 */
/**
 * @docs-group-description SearchBox
 * Components for working with a SearchBox.
 */
/**
 * @docs-group-description Select
 * Component is a wrapper for the `<select>` HTML element.
 */
/**
 * @docs-group-description Settings
 * Manager and UI Components that allow users to modify settings for different packages and extensions.
 */
/**
 * @docs-group-description Slider
 * Component displays a range slider with thumbs for changing the value.
 */
/**
 * @docs-group-description SplitButton
 * Components for working with a SplitButton.
 */
/**
 * @docs-group-description Tabs
 * Components for working with horizontal or vertical tabs.
 */
/**
 * @docs-group-description Text
 * Components for working with styled text.
 */
/**
 * @docs-group-description Tiles
 * Components for a container rendering elements that can be grouped together.
 */
/**
 * @docs-group-description Toggle
 * Components for working with a Toggle switch.
 */
/**
 * @docs-group-description Tooltip
 * Components for working with a Tooltip.
 */
/**
 * @docs-group-description Tree
 * Presentation React components for working with a Tree.
 */
/**
 * @docs-group-description UiStateStorage
 * Interfaces and classes for working with persistent UI settings.
 */
/**
 * @docs-group-description Utilities
 * Various utility classes, functions and React hooks for working with a UI.
 */
