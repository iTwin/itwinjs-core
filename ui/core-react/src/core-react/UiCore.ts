/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Utilities
 */

/** Import color variables, layout variables and Sass classes barrel file */
import "./colorthemes.scss";
import "./colorvariables.scss";
import "./layout-variables.scss";
import "./classes.scss";

import { Logger } from "@itwin/core-bentley";
import type { Localization } from "@itwin/core-common";
import { getClassName, UiError } from "@itwin/appui-abstract";

// cSpell:ignore colorthemes colorvariables

/**
 * Manages the Localization service for the core-react package.
 * @public
 */
export class UiCore {
  private static _initialized = false;
  private static _localization?: Localization;

  /**
   * Registers the Localization service namespace for UiCore.
   * @param localization The internationalization service created by the host application.
   */
  public static async initialize(localization: Localization): Promise<void> {
    if (UiCore._initialized) {
      Logger.logInfo(UiCore.loggerCategory(UiCore), `UiCore.initialize already called`);
      return;
    }

    UiCore._localization = localization;
    await UiCore._localization.registerNamespace(UiCore.localizationNamespace);
    UiCore._initialized = true;
  }

  /** Unregisters the UiCore localization namespace */
  public static terminate() {
    if (UiCore._localization)
      UiCore._localization.unregisterNamespace(UiCore.localizationNamespace);
    UiCore._localization = undefined;

    UiCore._initialized = false;
  }

  /** Determines if UiCore has been initialized */
  public static get initialized(): boolean { return UiCore._initialized; }

  /** The internationalization service created by the host application.
   * @internal
   */
  public static get localization(): Localization {
    // istanbul ignore else
    if (!UiCore._localization)
      throw new UiError(UiCore.loggerCategory(this), "localization: UiCore.initialize has not been called. Unable to return Localization object.");
    // istanbul ignore next
    return UiCore._localization;
  }

  /** The internationalization service namespace. */
  public static get localizationNamespace(): string {
    return "UiCore";
  }

  /** Calls localization.getLocalizedStringWithNamespace with the "UiCore" namespace. Do NOT include the namespace in the key.
   * @internal
   */
  public static translate(key: string | string[]): string {
    if (!UiCore._localization) {
      Logger.logError(UiCore.loggerCategory(this), `translate: UiCore must be initialize with a localization provider. Returning blank string.`);
      return "";
    }
    return UiCore._localization.getLocalizedStringWithNamespace(UiCore.localizationNamespace, key);
  }

  /** @internal */
  public static get packageName(): string {
    return "core-react";
  }

  /** @internal */
  public static loggerCategory(obj: any): string {
    const className = getClassName(obj);
    const category = UiCore.packageName + (className ? `.${className}` : "");
    return category;
  }

}
