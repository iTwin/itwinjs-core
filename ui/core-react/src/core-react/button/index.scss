/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "./button";

@import "./blue";
@import "./blue-large";
@import "./disabled";
@import "./disabled-large";
@import "./hollow";
@import "./hollow-large";
@import "./primary";
@import "./primary-large";
