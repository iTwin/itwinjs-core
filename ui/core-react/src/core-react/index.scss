/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@use "sass:math";

@import "./style/index";

@import "./_scrollbar";
@import "./_z-index";
@import "./_geometry";
@import "./_math";
@import "./_typography";

@import "./base/index";
@import "./button/index";
@import "./checkbox/index";
@import "./expandable/index";
@import "./icons/index";
@import "./inputs/index";
@import "./loading/index";
@import "./progress-indicators/index";
@import "./radio/index";
@import "./select/index";
@import "./tabs/index";
@import "./text/index";
@import "./tiles/index";
