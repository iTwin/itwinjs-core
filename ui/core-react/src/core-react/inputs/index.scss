/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
@import "../style/index";

@import "./variables";
@import "./input";
@import "./labeled-input";
@import "./labeled-textarea";
@import "./textarea";
