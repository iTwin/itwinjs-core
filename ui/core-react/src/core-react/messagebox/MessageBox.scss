/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
 @import "../icons/variables";
 @import "../style/colors";
 @import "../style/themecolors";
 @import "../style/baseline";

.core-message-box-container {
  display: flex;
  color: $buic-foreground-body;

  .core-message-box-icon {
    flex: 0 0;
    font-size: $uicore-icons-medium;
    padding: 0 $uicore-unbordered-padding $uicore-unbordered-padding $uicore-unbordered-padding;
    margin-top: 10px;
  }

  .core-message-box-content {
    display: flex;
    align-items: center;
    flex: 1 1;
  }
}

.core-message-box-success {
  color: $buic-foreground-success;
}
.core-message-box-information {
  color: $buic-foreground-primary;
}
.core-message-box-question {
  color: $uicore-yellow;
}
.core-message-box-warning {
  color: $buic-foreground-warning;
}
.core-message-box-error, .core-message-box-fatal {
  color: $buic-foreground-alert;
}

.notifymessage-icon {
  margin-right: 6px;
}
