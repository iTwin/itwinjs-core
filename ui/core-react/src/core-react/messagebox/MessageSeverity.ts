/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Dialog
 */

/** Message Severity enum.
  * @public
  * @deprecated in 3.0. Use [MessageSeverity]($appui-abstract) instead
  */
export enum MessageSeverity {
  None = 0,
  Information = 1,
  Question = 2,
  Warning = 3,
  Error = 4,
  Fatal = 5,
}
