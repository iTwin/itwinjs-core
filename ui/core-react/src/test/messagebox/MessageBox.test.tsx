/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
import { DialogButtonStyle, DialogButtonType, MessageSeverity } from "@itwin/appui-abstract";
import { render, screen } from "@testing-library/react";
import { expect } from "chai";
import * as React from "react";
import { MessageBox } from "../../core-react";
import { MessageContainer } from "../../core-react/messagebox/MessageBox";
import TestUtils, { classesFromElement } from "../TestUtils";

describe("MessageBox", () => {

  before(async () => {
    await TestUtils.initializeUiCore();
  });

  const buttonCluster = [
    { type: DialogButtonType.Close, buttonStyle: DialogButtonStyle.Primary, onClick: () => { } },
  ];

  describe("renders", () => {
    it("should render content in an open dialog", () => {
      render(<MessageBox opened={true} severity={MessageSeverity.Information} buttonCluster={buttonCluster}><div>Content</div></MessageBox>);

      expect(screen.getByText("Content", {selector: ".core-dialog.core-dialog-opened .core-message-box-container .core-message-box-content > div"})).to.exist;
    });
  });

  describe("renders different severities", () => {
    it("MessageSeverity.Question", () => {
      const {container} = render(<MessageBox opened={true} severity={MessageSeverity.Question} buttonCluster={buttonCluster} />);
      expect(classesFromElement(container.querySelector(".core-message-box-container")?.firstElementChild)).to.include("core-message-box-question");
    });
    it("MessageSeverity.Warning", () => {
      const {container} = render(<MessageBox opened={true} severity={MessageSeverity.Warning} buttonCluster={buttonCluster} />);
      expect(classesFromElement(container.querySelector(".core-message-box-container")?.firstElementChild)).to.include("core-message-box-warning");
    });
    it("MessageSeverity.Error", () => {
      const {container} = render(<MessageBox opened={true} severity={MessageSeverity.Error} buttonCluster={buttonCluster} />);
      expect(classesFromElement(container.querySelector(".core-message-box-container")?.firstElementChild)).to.include("core-message-box-error");
    });
    it("MessageSeverity.Fatal", () => {
      const {container} = render(<MessageBox opened={true} severity={MessageSeverity.Fatal} buttonCluster={buttonCluster} />);
      expect(classesFromElement(container.querySelector(".core-message-box-container")?.firstElementChild)).to.include("core-message-box-fatal");
    });
    it("MessageSeverity.None", () => {
      const {container} = render(<MessageBox opened={true} severity={MessageSeverity.None} buttonCluster={buttonCluster} />);
      expect(classesFromElement(container.querySelector(".core-message-box-container")?.firstElementChild)).to.not.include.members(
        ["icon-status-success-hollow" , "icon-status-success" , "core-message-box-success",
          "icon-info-hollow" , "icon-info" , "core-message-box-information",
          "icon-help-hollow" , "icon-help" , "core-message-box-question",
          "icon-status-warning" , "core-message-box-warning",
          "icon-status-error-hollow" , "icon-status-error" , "core-message-box-error",
          "icon-status-rejected" , "icon-status-rejected" , "core-message-box-fatal",
        ]
      );
    });
    it("MessageSeverity.Success", () => {
      const {container} = render(<MessageBox opened={true} severity={MessageSeverity.Success} buttonCluster={buttonCluster} />);
      expect(classesFromElement(container.querySelector(".core-message-box-container")?.firstElementChild)).to.include("core-message-box-success");
    });

  });

  describe("MessageContainer.getIconClassName with hollow param", () => {
    ([["None",MessageSeverity.None, " "],
      ["Information",MessageSeverity.Information, "icon-info-hollow"],
      ["Question",MessageSeverity.Question, "icon-help-hollow"],
      ["Warning",MessageSeverity.Warning, "icon-status-warning"],
      ["Error",MessageSeverity.Error, "icon-status-error-hollow"],
      ["Fatal",MessageSeverity.Fatal, "icon-status-rejected"],
      ["Success",MessageSeverity.Success, "icon-status-success-hollow"],
    ] as [string, MessageSeverity, string][]).map(([name, severity, className]) => {
      it(`hollow icon for ${name}`, () => {
        expect(MessageContainer.getIconClassName(severity, true)).to.include(className);
      });
    });
  });

});
