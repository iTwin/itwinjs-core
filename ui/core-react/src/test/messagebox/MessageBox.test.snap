// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox renders renders correctly 1`] = `
<Dialog
  alignment="center"
  buttonCluster={
    Array [
      Object {
        "buttonStyle": "iui-cta",
        "onClick": [Function],
        "type": "close",
      },
    ]
  }
  hideHeader={false}
  inset={true}
  minHeight={100}
  minWidth={300}
  modal={true}
  movable={false}
  opened={true}
  resizable={false}
  trapFocus={false}
  width="512px"
>
  <MessageContainer
    severity={1}
  />
</Dialog>
`;
